mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -497,40 +481,46 @@ final class OnlineLDAOptimizer extends LDAOptimizer with Logging {
       (u._1, u._2, u._3 + v._3)
     }
 
-    val (statsSum: BDM[Double], logphatOption: Option[BDV[Double]], nonEmptyDocsN: Long) = stats
-      .treeAggregate((BDM.zeros[Double](k, vocabSize), logphatPartOptionBase(), 0L))(
-        elementWiseSum, elementWiseSum
-      )
+    val (statsSum: BDM[Double], logphatOption: Option[BDV[Double]], batchSize: Long) =
+      batch.treeAggregate((BDM.zeros[Double](k, vocabSize), logphatPartOptionBase(), 0L))({
+        case (acc, (_, termCounts)) =>
+          val stat = BDM.zeros[Double](k, vocabSize)
```

Comment:
This is a per-record operation. Will it consume more memory than mapPartitions ? especially when k and vocabSize are large.

Reply:
Actually, we can fix this w/o falling back to `mapPartition`.